nixos/modules/services/games/vintagestory.nix

```
@@ -0,0 +1,118 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.vintagestory;
+in
+{
+  options.services.vintagestory = {
+    enable = lib.mkEnableOption "Vintagestory server";
+    package = lib.mkPackageOption pkgs "vintagestory" { };
+
+    openFirewall = lib.mkEnableOption "open the firewall on the specified port";
+
+    host = lib.mkOption {
+      description = "The address that the server will bind to";
+      default = null;
+      example = "127.0.0.1";
+      type = with lib.types; nullOr singleLineStr;
+    };
+
+    port = lib.mkOption {
+      description = "The port that the server will bind to";
+      default = 42420;
+      type = lib.types.ints.between 1 65535;
```

We have a type for that

```suggestion
      type = lib.types.port;
```